mozilla/fx-private-relay-add-on
# [MPP-1708] Popup Refactor - Register Subdomain / Create Custom Masks

## Summary

Links:
- Jira Tickets:
  - https://mozilla-hub.atlassian.net/browse/MPP-1770
  - https://mozilla-hub.atlassian.net/browse/MPP-1771
- L10N PR: https://github.com/mozilla-l10n/fx-private-relay-add-on-l10n/pull/44
- Epic PR #456 

Caveats/Not included: 
- Toggle/Expanded view of masks (to manage block level of masks) 
- Search bar/filter masks for premium masks

### Acceptance Criteria

[**MPP-1770**](https://mozilla-hub.atlassian.net/browse/MPP-1770)

- [x] Premium users only
- [x] CTA takes user to dashboard where the registration banner is shown (if a user has gone through onboarding)
- [X] ~CTA takes user to dashboard where onboarding screens are shown (if a user hasn’t gone through onboarding)~
  - Dev note: This is not possible.

[**MPP-1771**](https://mozilla-hub.atlassian.net/browse/MPP-1771)

- [x] Auto-prompt the site domain in the subdomain text field
- [x] Displays as the top mask
- [x] Displays open when you click into field your domain indicator disappears to allow more room to type

## Testing

[**MPP-1770 / Get Register Domain Prompt**](https://mozilla-hub.atlassian.net/browse/MPP-1770)

- Run add-on with premium account with subdomain NOT registered (and have no masks created) 
- Open panel
- **Expected:** There should be two prompts on the popup: one to create a random mask and a secondary button to register domain. 
- Click CTA
- **Expected:** The Relay dashboard page should open. (If on Firefox, the Register Subdomain banner/area should be highlighted) 
- Create random mask on dashboard
- Open popup
- **Expected:** There should be one mask (the same you just created) and a CTA to register subdomain
- Register a subdomain
- Open popup
- **Expected** The prompt should be gone

[**MPP-1771 / Create Custom Domain**](https://mozilla-hub.atlassian.net/browse/MPP-1771)

- Run add-on with premium account with subdomain registered
- Open panel
- **Expected:** There should be two mask generation buttons: "Create custom mask", "Create random mask"
- Click on "Create custom mask" button
- **Expected:** The create custom mask panel should be displayed and the mask name text input field should have focus
  - If on a website, the domain of the name (the _"mozilla"_ part of http://www.mozilla.org) will be pre-populated in the mask name text input field
  - If on an internal page (`about:`), the mask name text input field will display placeholder text
 - Fill out form with a mask name of "example123" and submit form
 - **Expected:** The mask should be created, and added to the top of the panel. 
- Click on "Create custom mask" button (again)
- Name this mask the same as previously, example123", and submit form
- **Expected** You should receive an error message. Click to dismiss


## Screenshots

Premium / No Masks Created / No Domain Registered

<img width="394" alt="image" src="https://user-images.githubusercontent.com/2692333/217073170-33d71890-4301-4b4c-bc05-e6c6fcd631f2.png">

Premium / Some Masks Created / No Domain Registered

<img width="408" alt="image" src="https://user-images.githubusercontent.com/2692333/217073223-8e2729da-60ca-4399-bace-342cdeef41aa.png">

Premium / Create Custom Mask Form (No mask name pre-populated)

<img width="403" alt="image" src="https://user-images.githubusercontent.com/2692333/217592029-b03cab85-3e48-4ebd-a844-5210c00d18be.png">

Premium / Create Custom Mask Form (Mask name is pre-populated)

<img width="399" alt="image" src="https://user-images.githubusercontent.com/2692333/217592008-1cf1216f-ea5e-489c-b9ad-7794a3e30f90.png">

Premium / Create Custom Mask Form (Tooltip of Promotional Blocking)

<img width="388" alt="image" src="https://user-images.githubusercontent.com/2692333/217592057-72259e83-9fcc-4e38-97e4-c96e0b303028.png">

Premium / Create Custom Mask Form (Error making mask)

<img width="389" alt="image" src="https://user-images.githubusercontent.com/2692333/217593737-e42684e8-9318-4c13-a40a-0f9b4bf4358e.png">
